R/length-params.R

```diff
@@ -0,0 +1,155 @@
+#' Download metadata catch types
+#'
+#' This function download airtable tables containing metadata of catch types.
+#'
+#' @param log_threshold The (standard Apache logj4) log level used as a
+#'   threshold for the logging infrastructure. See [logger::log_levels] for more
+#'   details
+#'
+#' @return No output. This funcrion is used for it's side effects
+#' @export
+#'
+get_catch_types <- function(log_threshold = logger::DEBUG) {
+  pars <- read_config()
+
+  metadata <- get_preprocessed_metadata(pars)
+
+  catch_types <- metadata$catch_types
+  fao_records <- metadata$fao_catch
+
+  catches_tab <- dplyr::left_join(fao_records, catch_types, by = "interagency_code")
+
+  spp <- catches_tab %>%
+    dplyr::select(
+      .data$interagency_code, .data$isscaap_taxocode.x, .data$name_scientific,
+      .data$catch_family, .data$catch_name_en, .data$taxo_name_en
+    )
+
+  logger::log_info("Getting taxa ranks...")
+  ranks <- taxize::tax_rank(spp$name_scientific, db = "gbif", rows = 1)
+
+
+  ranks_tab <-
+    ranks %>%
+    purrr::map(~ stringr::str_to_title(.x)) %>%
+    purrr::map(~ tidyr::as_tibble(.x)) %>%
+    purrr::map_dfr(~ dplyr::mutate_all(.x, as.character)) %>%
+    dplyr::rename(taxa_rank = .data$value) %>%
+    dplyr::bind_cols(spp) %>%
+    dplyr::mutate(name = gsub(" spp", "", .data$name_scientific)) %>%
+    dplyr::mutate(name = gsub("\\s*\\([^\\)]+\\)", "", .data$name)) %>%
+    dplyr::mutate(
+      name = dplyr::case_when(
+        name == "Thunnini" ~ "Tuna",
+        name == "Selachimorpha" ~ "Shark",
+        TRUE ~ .data$name
+      ),
+      taxa_rank = dplyr::case_when(
+        name == "Tuna" ~ "comm_name",
+        name == "Shark" ~ "comm_name",
+        TRUE ~ .data$taxa_rank
+      )
+    ) %>%
+    dplyr::filter(!.data$catch_name_en %in% c(
+      "No catch", "Sea cucumber",
+      "Other", "Unknown", "Lobster",
+      "Herring"
```

Comment:
Perhaps we should also exclude octopus, shrimps, cuttlefish, crabs, cockles, and algae? Also wondering why are herrings excluded here?